docs/spec/v1beta2/gitrepositories.md

```diff
@@ -287,6 +287,25 @@ spec:
 This field takes precedence over [`.branch`](#branch-example) and
 [`.tag`](#tag-example).
 
+
+#### Name example
+
+To Git checkout to a specfic reference, use `.spec.ref.name`:
+
+```yaml
+---
+apiVersion: source.toolkit.fluxcd.io/v1beta2
+kind: GitRepository
+metadata:
+  name: <repository-name>
+spec:
+  ref:
+    # Ref name format reference: https://git-scm.com/docs/git-check-ref-format#_description
+    name: <ref name>
+```
+
+Valid examples are: `refs/heads/main`, `refs/tags/v0.1.0`, `refs/pull/420/head`, `refs/merge-requests/1/head`
```

Comment:
```suggestion
Valid examples are: `refs/heads/main`, `refs/tags/v0.1.0`, `refs/pull/420/head`,
`refs/merge-requests/1/head`.

This field takes precedence over [`.branch`](#branch-example),
[`.tag`](#tag-example), and [`.semver`](#semver example).
```

Reply:
didn't mention the order as that's already specified here: https://fluxcd.io/flux/components/source/gitrepositories/#reference

Reply:
See the other sections, the reason we mention it per section as well is because people click on links with a `#ref` and then do not read the text above.